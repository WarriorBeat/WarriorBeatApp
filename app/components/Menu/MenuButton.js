/**
 * MenuButton.js
 * MenuButton Component
 * components/Menu
 */

import React from "react"
import PropTypes from "prop-types"
import { Button } from "react-native-elements"
import { ViewPropTypes } from "react-native"
import sideMenu from "./styles"

const MenuButton = (props) => {
  const {
    isHeader, isFooter, title, icon, onPress, buttonStyle, titleStyle, requiresAuth,
  } = props
  let btnTitleStyle = isHeader ? sideMenu.buttonHeaderText : sideMenu.buttonText
  btnTitleStyle = isFooter ? sideMenu.buttonFooterText : btnTitleStyle

  const btnStyle = isFooter ? sideMenu.buttonSecondary : sideMenu.button
  if (requiresAuth !== false || requiresAuth === undefined) {
    return (
      <Button
        type="clear"
        titleStyle={{ ...btnTitleStyle, ...titleStyle }}
        buttonStyle={{ ...btnStyle, ...buttonStyle }}
        title={title}
        icon={{ ...icon, color: "white" }}
        onPress={onPress}
      />
    )
  }
  return null
}

MenuButton.propTypes = {
  isHeader: PropTypes.bool,
  isFooter: PropTypes.bool,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  icon: PropTypes.objectOf(PropTypes.string),
  onPress: PropTypes.func,
  buttonStyle: ViewPropTypes.style,
  titleStyle: ViewPropTypes.style,
  requiresAuth: ViewPropTypes.bool,
}

MenuButton.defaultProps = {
  isHeader: false,
  isFooter: false,
  icon: {},
  onPress: () => {},
  buttonStyle: {},
  titleStyle: {},
  requiresAuth: null,
}

export default MenuButton
