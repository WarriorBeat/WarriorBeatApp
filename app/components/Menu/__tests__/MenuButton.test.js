/**
 * __tests__/MenuButton.test.js
 * Component Test
 * tests
 */
/* eslint-disable */

import React from "react"
import { render } from "react-native-testing-library"
import { HeaderButton, FooterButton } from "../MenuButton"

test("should render HeaderButton correctly", () => {
  const tree = render(<HeaderButton title="Button" icon={{ name: "home", type: "Entypo" }} />)
  expect(tree).toMatchSnapshot()
})

test("should render FooterButton correctly", () => {
  const tree = render(<FooterButton title="Button" icon={{ name: "home", type: "Entypo" }} />)
  expect(tree).toMatchSnapshot()
})
