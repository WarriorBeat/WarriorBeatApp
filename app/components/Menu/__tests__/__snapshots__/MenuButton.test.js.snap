// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FooterButton correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": null,
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 0,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#f4f4f4",
            "fontSize": 24,
            "marginLeft": 16,
            "marginRight": 0,
            "paddingTop": 2,
          },
        ],
        Object {
          "fontFamily": "Entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    type="Entypo"
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#f4f4f4",
          "fontFamily": "System",
          "fontSize": 22,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Object {
          "color": "rgba(255,255,255,0.70)",
          "fontSize": 16,
          "fontWeight": "bold",
        },
      ]
    }
    uppercase={false}
  >
    Button
  </Text>
</View>
`;

exports[`should render HeaderButton correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": null,
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 0,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#f4f4f4",
            "fontSize": 24,
            "marginLeft": 16,
            "marginRight": 0,
            "paddingTop": 2,
          },
        ],
        Object {
          "fontFamily": "Entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    type="Entypo"
  >
    
  </Text>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#f4f4f4",
          "fontFamily": "System",
          "fontSize": 22,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Object {
          "color": "rgba(255,255,255,1)",
          "fontWeight": "bold",
        },
      ]
    }
    uppercase={false}
  >
    Button
  </Text>
</View>
`;
